### PR TITLE
Add in `confirm-mark-feed-read` configuration variable

### PR DESCRIPTION
**Pull request by aflyingpumpkin**
This adds in a `confirm-mark-feed-read` using code from `confirm-mark-all-feeds-read`
Fixes: #1781

### CODE REVIEW

File: src/configcontainer.cpp [hunk 1]
```diff
@@ -46,6 +46,7 @@ ConfigContainer::ConfigContainer()
 	{"cleanup-on-quit", ConfigData("yes", ConfigDataType::BOOL)},
 	{"confirm-delete-all-articles", ConfigData("yes", ConfigDataType::BOOL)},
 	{"confirm-mark-all-feeds-read", ConfigData("yes", ConfigDataType::BOOL)},
+	{"confirm-mark-feed-read", ConfigData("no", ConfigDataType::BOOL)},
```
Comment (Minoru) [inline src/configcontainer.cpp:49]: Please document this setting in _doc/configcommands.dsv_ (note that the file is sorted alphabetically).

Also, please change the default to `yes`. Good on you for caring about backwards compatibility, but I think the benefits of confirmation outweigh the costs for the existing users.

File: src/itemlistformaction.cpp [hunk 2]
```diff
@@ -482,56 +482,61 @@ bool ItemListFormAction::process_operation(Operation op,
...
-				bool notify = visible_items[0].first->feedurl() != feed->rssurl();
+		if (!cfg->get_configvalue_as_bool(
+				"confirm-mark-feed-read") ||
+			v->confirm(_("Do you really want to mark this feed as read (y:Yes n:No)? "),
+				_("yn")) == *_("y")) {
+			LOG(Level::INFO, "ItemListFormAction: marking feed read");
+			try {
+				const auto message_lifetime = v->get_statusline().show_message_until_finished(
+						_("Marking feed read..."));
+
```
Comment (Minoru) [inline src/itemlistformaction.cpp:493]: Apparently this and other empty lines contain a tab; please remove those tabs.

**Review by Minoru — CHANGES_REQUESTED** [review]
Thanks!

File: src/itemlistformaction.cpp (hunk 2 above)
Comment (aflyingpumpkin) [reply src/itemlistformaction.cpp]: I believe this has been fixed.

File: src/configcontainer.cpp (hunk 1 above)
Comment (aflyingpumpkin) [reply src/configcontainer.cpp]: Fixed!

File: src/itemlistformaction.cpp (hunk 2 above)
Comment (aflyingpumpkin) [reply src/itemlistformaction.cpp]: Oh, never mind I just realized what you were talking about. I will actually resolve it shortly.

Comment (Minoru) [reply src/itemlistformaction.cpp]: Doesn't look like it: https://github.com/newsboat/newsboat/pull/1782/checks?check_run_id=3655285773

**Comment by Minoru** [comment]
Could you please rebase this onto master instead of merging? It's a bit of a mess, with your changes mixed into stuff that's already in master. I probably won't even be able to squash it.

**Comment by Minoru** [comment]
> Would the best way to go about this be redoing the changes in a new pull request that has not been messed up?

Let's try to fix this one instead :) There aren't that many changes, so we can do as follows:

1. look at your first commit in this PR, 2194f927e1f7a6685a9547ea8cb51c31dc8c8c89, and see who its parent is: `git show 2194f927e1f7a6685a9547ea8cb51c31dc8c8c89^` (the caret, `^`, means "parent"). The result is b76911029d5df7b6e8443cebaa5221181b2fb38b.
2. roll back to that parent, *keeping all the changes made since then*: `git reset --soft b76911029d5df7b6e8443cebaa5221181b2fb38b`. If you run `git status now`, you'll see a lot of modified files, including README.md and others — that's fine.
3. commit as if all those changes are yours: `git commit -m'added confirm-mark-feed-read config variable '`. We basically squashed this entire PR into a single commit.
4. now the final step: rebase our "squashed" commit onto the current master: `git rebase master`. This will drop any changes that are already in master, leaving just the ones that *you* made. You can verify the result with `git show`.
5. (now would be a good time to fix that last remaining style error: https://github.com/newsboat/newsboat/pull/1782/checks?check_run_id=3655526750)
6. push your changes: `git push --force`. You have to use `--force` in this case because you're overwriting this PR with the new commit. Normally force-pushes to PRs are frowned upon, because they make me re-review the entire PR, but this one is small enough that it's okay.

That should do it. For future PRs, create a separate branch and don't merge upstream's master into it.

(If the above doesn't work, just tell me, I'll try to fix this myself. Also, this is just one of the ways to do it, the one that I felt was the most appropriate.)

File: src/itemlistformaction.cpp (hunk 2 above)
Comment (aflyingpumpkin) [reply src/itemlistformaction.cpp]: Should finally be fixed, or at the very least the automatic format checker is not yelling at me anymore.

**Review by Minoru — APPROVED** [review]
Thanks! :+1:

**Pull request merged**